drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -29,39 +29,25 @@ local COLOR_TEMPERATURE_KELVIN_MIN = 1
 local COLOR_TEMPERATURE_MIRED_MAX = CONVERSION_CONSTANT/COLOR_TEMPERATURE_KELVIN_MIN
 local COLOR_TEMPERATURE_MIRED_MIN = CONVERSION_CONSTANT/COLOR_TEMPERATURE_KELVIN_MAX
 
+local ENDPOINT_TO_COMPONENT_MAP = "__endpoint_to_component_map"
+
 local function convert_huesat_st_to_matter(val)
   return math.floor((val * 0xFE) / 100.0 + 0.5)
 end
 
---- component_to_endpoint helper function to handle situations where
---- device does not have endpoint ids in sequential order from 1
---- In this case the function returns the lowest endpoint value that isn't 0
-local function find_default_endpoint(device, component)
-  local res = device.MATTER_DEFAULT_ENDPOINT
-  local eps = device:get_endpoints(nil)
-  table.sort(eps)
-  for _, v in ipairs(eps) do
-    if v ~= 0 then --0 is the matter RootNode endpoint
-      res = v
-      break
-    end
+local function component_to_endpoint(device, component_name)
+  local map = device:get_field(ENDPOINT_TO_COMPONENT_MAP) or {}
+  for ep, component in pairs(map) do
+    if component == component_name then return ep end
   end
-  return res
-end
-
-local function component_to_endpoint(device, component_id)
-  -- Assumes matter endpoint layout is sequentional starting at 1.
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) or find_default_endpoint(device, component_id)
 end
```

Comment:
Do we need a default return value here?

Reply:
I think so. I think this will break existing devices that dont have an`ENDPOINT_TO_COMPONENT_MAP` setup on the device object. Previously it would have returned the lowest non-0 endpoint, but now it will return nil

Reply:
Added a default return of the lowest non-0 endpoint in a9e0c3ac0e37798c0fd3f4927ce6b4ad80d88fc0

Reply:
the matter device's default behavior is to return the value of `find_default_endpoint` if the `component_to_endpoint` function doesn't return anything, so it's not technically necessary

Reply:
> the matter device's default behavior is to return the value of find_default_endpoint if the component_to_endpoint function doesn't return anything, so it's not technically necessary

I think it only defaults to `find_default_endpoint` if the `component_to_endpoint` function itself is nil (i.e. has not been set by the driver), not if the `component_to_endpoint` function returns nil